src/Zicht/Bundle/UrlBundle/Validator/Constraints/ContainsUniquePublicUrlValidator.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * @author    Philip Bergman <philip@zicht.nl>
+ * @copyright Zicht Online <http://www.zicht.nl>
+ */
+namespace Zicht\Bundle\UrlBundle\Validator\Constraints;
+
+use Doctrine\Bundle\DoctrineBundle\Registry;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * Class ContainsUniquePublicUrlValidator
+ *
+ * @package Zicht\Bundle\UrlBundle\Validator\Constraints
+ */
+class ContainsUniquePublicUrlValidator extends ConstraintValidator
+{
+    /** @var Registry  */
+    protected $doctrine;
+
+    /**
+     * Constructor
+     *
+     * @param Registry $doctrine
+     */
+    public function __construct(Registry $doctrine)
+    {
+        $this->doctrine = $doctrine;
+    }
+
+
+    /**
+     * Checks if the passed value is valid.
+     *
+     * @param mixed $value The value that should be validated
+     * @param Constraint $constraint The constraint for the validation
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (0 < $this->getPublicUrlCount($value)) {
+            $this->context->addViolation($constraint->message, ['%url%' => $value]);
+        }
+    }
+
+    /**
+     * @param string $url
+     * @return int
+     */
+    protected function getPublicUrlCount($url)
+    {
+        return (int)$this->doctrine->getConnection()->query($this->fmtQuery($url))->fetchColumn(0);
+    }
+
+    /**
+     * Create a select (count) query with the given url.
+     *
+     * @param string $url
+     * @return string
+     */
+    protected function fmtQuery($url)
+    {
+        return sprintf('SELECT COUNT(*) FROM url_alias WHERE public_url = %s', $this->doctrine->getConnection()->quote($url));
```

Comment:
dit is niet goed. Je quote weliswaar via de connection maar je  concatineert het wel zelf. je zou het volledig over moeten laten aan mysql door prepared statements te gebruiken. dit kan ervoor zorgen dat iemand via een url een sql injectie kan doen. dat kan bijvoorbeeld door dubbel quoting zwakheden in escaping technieken.

Reply:
Waar heb je dat vandaan? Volgens hun eigen docs is dit gewoon goed. 

http://docs.doctrine-project.org/projects/doctrine-dbal/en/latest/reference/security.html#right-quoting-escaping-values

Reply:
het is toch een beetje vreemd om voor de slechtste alternatief te gaan? Prepared statements zijn beter. die hebben ook vele voordelen. een concattenatie is heel fout gevoelig in sql. waarom kies je voor een slechte alternatief als je vele betere hebt?

Reply:
Ik ben wel benieuwed waarom ze beter zijn en waar je dat van daan heb gehaald? Maar zal er verder kort over zijn, ik gebruik dit omdat raw queries gebruikt.

Reply:
Misschien niet de meest perfecte oplossing volgens het boekje, maar het werkt prima en er is zo geen security probleem.  Een prepared statement is misschien netter maar ook iets minder efficient.  Dat gevecht ga ik niet aan, dit is wat mij betreft prima.